test/decl/nested/protocol.swift

```diff
@@ -115,15 +257,55 @@ func testLookup<T: OuterForUFI.Inner>(_ x: T) {
   x.extMethod()
 }
 
-// rdar://problem/113103854
+// Protocols cannot be nested inside of generic functions.
 
-protocol Q {
-  associatedtype A
+func invalidProtocolInGeneric<T>(_: T) {
+  protocol Test {} // expected-error{{type 'Test' cannot be nested in generic function 'invalidProtocolInGeneric'}}
 }
 
-struct OuterNonGeneric {
-  protocol P: Q where Self.A == Int {}
-  // expected-error@-1 {{protocol 'P' cannot be nested inside another declaration}}
+struct NestedInGenericMethod<T> {
+  func someMethod() {
+    protocol AnotherTest {} // expected-error{{type 'AnotherTest' cannot be nested in generic function 'someMethod()'}}
```

Comment:
Shouldn't this use the "protocol %0 cannot be nested in a generic context" diagnostic?

Reply:
Another change (that I rebased in) guarded these checks inside `if (auto *parentDecl = DC->getSelfNominalTypeDecl())`. In the function, `getSelfNominalTypeDecl` returns null, so this goes along the regular nominal-in-generic-context error path. That path gives nicer diagnostics, so I left it that way.